src/send/usePrepareSendTransactions.ts

```diff
@@ -1,35 +1,138 @@
 import BigNumber from 'bignumber.js'
 import { useAsyncCallback } from 'react-async-hook'
+import erc20 from 'src/abis/IERC20'
+import jumpstart from 'src/abis/IWalletJumpstart'
+import { getDynamicConfigParams } from 'src/statsig'
+import { DynamicConfigs } from 'src/statsig/constants'
+import { StatsigDynamicConfigs } from 'src/statsig/types'
 import { tokenAmountInSmallestUnit } from 'src/tokens/saga'
 import { TokenBalance, isNativeTokenBalance, tokenBalanceHasAddress } from 'src/tokens/slice'
 import { tokenSupportsComments } from 'src/tokens/utils'
+import { NetworkId } from 'src/transactions/types'
 import Logger from 'src/utils/Logger'
+import { publicClient } from 'src/viem'
 import {
+  TransactionRequest,
   prepareERC20TransferTransaction,
   prepareSendNativeAssetTransaction,
+  prepareTransactions,
   prepareTransferWithCommentTransaction,
 } from 'src/viem/prepareTransactions'
+import { networkIdToNetwork } from 'src/web3/networkConfig'
+import { Address, encodeFunctionData } from 'viem'
 
 const TAG = 'src/send/usePrepareSendTransactions'
 
-export async function prepareSendTransactionsCallback({
-  amount,
-  token,
-  recipientAddress,
-  walletAddress,
-  feeCurrencies,
-  comment,
-}: {
+async function createBaseJumpstartTransactions(
+  jumpstartContractAddress: string,
+  spendTokenAmount: BigNumber,
+  spendTokenAddress: string,
+  networkId: NetworkId,
+  walletAddress: string,
+  publicKey: string
+) {
+  const baseTransactions: TransactionRequest[] = []
+  const spendAmount = BigInt(spendTokenAmount.toFixed(0))
+
+  const approvedAllowanceForSpender = await publicClient[
+    networkIdToNetwork[networkId]
+  ].readContract({
+    address: spendTokenAddress as Address,
+    abi: erc20.abi,
+    functionName: 'allowance',
+    args: [walletAddress as Address, jumpstartContractAddress as Address],
+  })
+
+  if (approvedAllowanceForSpender < spendAmount) {
+    const approveTx: TransactionRequest = {
+      from: walletAddress as Address,
+      to: spendTokenAddress as Address,
+      data: encodeFunctionData({
+        abi: erc20.abi,
+        functionName: 'approve',
+        args: [jumpstartContractAddress as Address, spendAmount],
+      }),
+    }
+    baseTransactions.push(approveTx)
+  }
+
+  const transferTx: TransactionRequest = {
+    from: walletAddress as Address,
+    to: jumpstartContractAddress as Address,
+    value: BigInt(0),
```

Comment:
i wonder why we are specifying `value` for this tx, but not doing the same for th e `approveTx`?